exchanges/fee/fee.go

```diff
@@ -0,0 +1,74 @@
+package fee
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/thrasher-corp/gocryptotrader/common/key"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+var (
+	// ErrExchangeNameEmpty is an error for when the exchange name is empty
+	ErrExchangeNameEmpty = errors.New("exchange name is empty")
+	// ErrFeeRateNotFound is an error for when the fee rate is not found
+	ErrFeeRateNotFound = errors.New("fee rate not found")
+	exchangeFees       = Fees{all: make(map[key.ExchangePairAsset]Rates)}
+)
+
+// Fees holds the fees for each exchange.
+type Fees struct {
+	all map[key.ExchangePairAsset]Rates
```

Comment:
Just more a commentary on fees, withdrawal and deposit fees are typically by currency code. I would say that this system is for `TradingFees` however, could also end up being adapted for deposit and withdrawal fees too. So in that sense, I think `all` -> `tradingFees`